src/librustdoc/html/render/mod.rs

```diff
@@ -4283,6 +4282,7 @@ fn small_url_encode(s: String) -> String {
 
         st += &s[last_match..idx];
         st += escaped;
+        // NOTE: we only handle single byte characters here
```

Comment:
That's not great and could be very problematic... I see two solutions here: either you use `last_match = idx + c.len_utf8()`, but I'm not sure if it'll always be utf8, or you use `peekable` like:

```rust
let mut iter = s.char_indices().peekable();
while let Some((idx, c)) = iter.next() {
    // ...
    last_match = idx + iter.peek().map(|(i, _)| i).unwrap_or(0);
```

I tend to prefer the second case.

Reply:
I mean, we skip this if the character is not in the match list above...

Reply:
Oh I see! You're right.